xapian-core/common/parseint.h

```diff
@@ -37,4 +38,21 @@ bool parse_unsigned(const char* p, T& res)
     return true;
 }
 
+template<typename T>
+bool parse_signed(const char* p, T& res)
+{
+    typedef typename std::make_unsigned<T>::type unsigned_type;
+    unsigned_type temp = 0;
+    if (*p == '-' && parse_unsigned(++p, temp) &&
+	temp <= unsigned_type(std::numeric_limits<T>::min())) {
+	res = -temp;
+	return true;
+    } else if (parse_unsigned(p, temp) &&
+	temp <= std::numeric_limits<T>::max()) {
```

Comment:
The `t` of `temp` should be aligned under `p` of `parse_unsigned` here.

Reply:
fixed in a fixup commit